Failing test cases

```
======================================================= test session starts =======================================================
platform linux -- Python 3.6.13, pytest-3.6.1, py-1.10.0, pluggy-0.6.0
rootdir: /home/chris/dev/git_work/distil-primitives, inifile: setup.cfg
plugins: timeout-1.3.0, pythonpath-0.7.1, cov-2.5.1, Faker-8.8.1
collected 38 items

test/test_annotation.py .                                                                                                   [  2%]
test/test_audio_transfer.py FFF                                                                                             [ 10%]
test/test_binary_encoder.py ..                                                                                              [ 15%]
test/test_categorical_imputer.py ......                                                                                     [ 31%]
test/test_column_parser.py ......                                                                                           [ 47%]
test/test_image_transfer.py ..                                                                                              [ 52%]
test/test_list_to_dataframe.py ...                                                                                          [ 60%]
test/test_one_hot_encoder.py ..                                                                                             [ 65%]
test/test_ranked_linear_svc.py ....                                                                                         [ 76%]
test/test_replace_singletons.py .                                                                                           [ 78%]
test/test_satellite_image_loader.py ...                                                                                     [ 86%]
test/test_text_encoder.py FFF                                                                                               [ 94%]
test/test_timeseries_formatter.py ..                                                                                        [100%]

============================================================ FAILURES =============================================================
____________________________________________ AudioTransferPrimitiveTestCase.test_basic ____________________________________________

self = Index(['d3mIndex', 'filename', 'start', 'end', 'class'], dtype='object'), key = 'start-end-time-slice-of-recording'
method = None, tolerance = None

    def get_loc(self, key, method=None, tolerance=None):
        """
            Get integer location, slice or boolean mask for requested label.

            Parameters
            ----------
            key : label
            method : {None, 'pad'/'ffill', 'backfill'/'bfill', 'nearest'}, optional
                * default: exact matches only.
                * pad / ffill: find the PREVIOUS index value if no exact match.
                * backfill / bfill: use NEXT index value if no exact match
                * nearest: use the NEAREST index value if no exact match. Tied
                  distances are broken by preferring the larger index value.
            tolerance : int or float, optional
                Maximum distance from index value for inexact matches. The value of
                the index at the matching location most satisfy the equation
                ``abs(index[loc] - key) <= tolerance``.

            Returns
            -------
            loc : int if unique index, slice if monotonic index, else mask

            Examples
            --------
            >>> unique_index = pd.Index(list('abc'))
            >>> unique_index.get_loc('b')
            1

            >>> monotonic_index = pd.Index(list('abbc'))
            >>> monotonic_index.get_loc('b')
            slice(1, 3, None)

            >>> non_monotonic_index = pd.Index(list('abcb'))
            >>> non_monotonic_index.get_loc('b')
            array([False,  True, False,  True])
            """
        if method is None:
            if tolerance is not None:
                raise ValueError(
                    "tolerance argument only valid if using pad, "
                    "backfill or nearest lookups"
                )
            casted_key = self._maybe_cast_indexer(key)
            try:
>               return self._engine.get_loc(casted_key)

../../miniconda3/envs/distil-auto-ml/lib/python3.6/site-packages/pandas/core/indexes/base.py:2898:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

>   ???

pandas/_libs/index.pyx:70:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

>   ???

pandas/_libs/index.pyx:101:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

>   ???

pandas/_libs/hashtable_class_helper.pxi:1675:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

>   ???
E   KeyError: 'start-end-time-slice-of-recording'

pandas/_libs/hashtable_class_helper.pxi:1683: KeyError

The above exception was the direct cause of the following exception:

self = <test_audio_transfer.AudioTransferPrimitiveTestCase testMethod=test_basic>

    def test_basic(self):
        dataset = test_utils.load_dataset(self._dataset_path)

        audio_loader_hyperparams = (
            AudioDatasetLoaderPrimitive.metadata.get_hyperparams()
        )
        audio_loader_primitive = AudioDatasetLoaderPrimitive(
            hyperparams=audio_loader_hyperparams.defaults()
        )
>       audio_df = audio_loader_primitive.produce_collection(inputs=dataset).value

test/test_audio_transfer.py:27:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
distil/primitives/audio_reader.py:244: in produce_collection
    start_end = row["start-end-time-slice-of-recording"]
../../miniconda3/envs/distil-auto-ml/lib/python3.6/site-packages/pandas/core/series.py:882: in __getitem__
    return self._get_value(key)
../../miniconda3/envs/distil-auto-ml/lib/python3.6/site-packages/pandas/core/series.py:990: in _get_value
    loc = self.index.get_loc(label)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = Index(['d3mIndex', 'filename', 'start', 'end', 'class'], dtype='object'), key = 'start-end-time-slice-of-recording'
method = None, tolerance = None

    def get_loc(self, key, method=None, tolerance=None):
        """
            Get integer location, slice or boolean mask for requested label.

            Parameters
            ----------
            key : label
            method : {None, 'pad'/'ffill', 'backfill'/'bfill', 'nearest'}, optional
                * default: exact matches only.
                * pad / ffill: find the PREVIOUS index value if no exact match.
                * backfill / bfill: use NEXT index value if no exact match
                * nearest: use the NEAREST index value if no exact match. Tied
                  distances are broken by preferring the larger index value.
            tolerance : int or float, optional
                Maximum distance from index value for inexact matches. The value of
                the index at the matching location most satisfy the equation
                ``abs(index[loc] - key) <= tolerance``.

            Returns
            -------
            loc : int if unique index, slice if monotonic index, else mask

            Examples
            --------
            >>> unique_index = pd.Index(list('abc'))
            >>> unique_index.get_loc('b')
            1

            >>> monotonic_index = pd.Index(list('abbc'))
            >>> monotonic_index.get_loc('b')
            slice(1, 3, None)

            >>> non_monotonic_index = pd.Index(list('abcb'))
            >>> non_monotonic_index.get_loc('b')
            array([False,  True, False,  True])
            """
        if method is None:
            if tolerance is not None:
                raise ValueError(
                    "tolerance argument only valid if using pad, "
                    "backfill or nearest lookups"
                )
            casted_key = self._maybe_cast_indexer(key)
            try:
                return self._engine.get_loc(casted_key)
            except KeyError as err:
>               raise KeyError(key) from err
E               KeyError: 'start-end-time-slice-of-recording'

../../miniconda3/envs/distil-auto-ml/lib/python3.6/site-packages/pandas/core/indexes/base.py:2900: KeyError
_______________________________ AudioTransferPrimitiveTestCase.test_no_hyperparams_no_semantic_type _______________________________

self = <test_audio_transfer.AudioTransferPrimitiveTestCase testMethod=test_no_hyperparams_no_semantic_type>

    def test_no_hyperparams_no_semantic_type(self):
        dataset = test_utils.load_dataset(self._dataset_path)

        audio_loader_hyperparams = (
            AudioDatasetLoaderPrimitive.metadata.get_hyperparams()
        )
        audio_loader_primitive = AudioDatasetLoaderPrimitive(
            hyperparams=audio_loader_hyperparams
        )
>       audio_df = audio_loader_primitive.produce(inputs=df).value
E       NameError: name 'df' is not defined

test/test_audio_transfer.py:82: NameError
________________________________ AudioTransferPrimitiveTestCase.test_no_hyperparams_semantic_type _________________________________

self = <test_audio_transfer.AudioTransferPrimitiveTestCase testMethod=test_no_hyperparams_semantic_type>

    def test_no_hyperparams_semantic_type(self):
        dataset = test_utils.load_dataset(self._dataset_path)

        dataframe_hyperparams_class = (
            dataset_to_dataframe.DatasetToDataFramePrimitive.metadata.get_hyperparams()
        )
        dataframe_primitive = dataset_to_dataframe.DatasetToDataFramePrimitive(
            hyperparams=dataframe_hyperparams_class.defaults()
        )
        df = dataframe_primitive.produce(inputs=dataset).value

>       audio_df.metadata = audio_df.metadata.add_semantic_type(
            (metadata_base.ALL_ELEMENTS, 0), "http://schema.org/AudioObject"
        )
E       NameError: name 'audio_df' is not defined

test/test_audio_transfer.py:56: NameError
__________________________________ TextEncoderPrimitiveTestCase.test_classification_binary_label __________________________________

self = <test_text_encoder.TextEncoderPrimitiveTestCase testMethod=test_classification_binary_label>

    def test_classification_binary_label(self) -> None:
        # load test data into a dataframe
        dataset = test_utils.load_dataset(self._dataset_path)
        dataframe = test_utils.get_dataframe(dataset, "learningData")
        dataframe = dataframe.iloc[0:5]

        # create the encoder
        hyperparams_class = TextEncoderPrimitive.metadata.query()["primitive_code"][
            "class_type_arguments"
        ]["Hyperparams"]
        encoder = TextEncoderPrimitive(hyperparams=hyperparams_class.defaults())
        encoder.set_training_data(
>           inputs=dataframe.iloc[:, [0, 1]], outputs=dataframe.iloc[:, 2]
        )

test/test_text_encoder.py:91:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
distil/primitives/text_encoder.py:124: in set_training_data
    outputs = outputs.iloc[:, 0].values
../../miniconda3/envs/distil-auto-ml/lib/python3.6/site-packages/pandas/core/indexing.py:873: in __getitem__
    return self._getitem_tuple(key)
../../miniconda3/envs/distil-auto-ml/lib/python3.6/site-packages/pandas/core/indexing.py:1443: in _getitem_tuple
    self._has_valid_tuple(tup)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <pandas.core.indexing._iLocIndexer object at 0x7f94183e8728>, key = (slice(None, None, None), 0)

    def _has_valid_tuple(self, key: Tuple):
        """
            Check the key for valid keys across my indexer.
            """
        for i, k in enumerate(key):
            if i >= self.ndim:
>               raise IndexingError("Too many indexers")
E               pandas.core.indexing.IndexingError: Too many indexers

../../miniconda3/envs/distil-auto-ml/lib/python3.6/site-packages/pandas/core/indexing.py:700: IndexingError
________________________________ TextEncoderPrimitiveTestCase.test_classification_singleton_label _________________________________

self = <test_text_encoder.TextEncoderPrimitiveTestCase testMethod=test_classification_singleton_label>

    def test_classification_singleton_label(self) -> None:
        # load test data into a dataframe
        dataset = test_utils.load_dataset(self._dataset_path)
        dataframe = test_utils.get_dataframe(dataset, "learningData")
        dataframe = dataframe.iloc[0:6]

        # create the encoder
        hyperparams_class = TextEncoderPrimitive.metadata.query()["primitive_code"][
            "class_type_arguments"
        ]["Hyperparams"]
        encoder = TextEncoderPrimitive(hyperparams=hyperparams_class.defaults())
        encoder.set_training_data(
>           inputs=dataframe.iloc[:, [0, 1]], outputs=dataframe.iloc[:, 2]
        )

test/test_text_encoder.py:121:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
distil/primitives/text_encoder.py:124: in set_training_data
    outputs = outputs.iloc[:, 0].values
../../miniconda3/envs/distil-auto-ml/lib/python3.6/site-packages/pandas/core/indexing.py:873: in __getitem__
    return self._getitem_tuple(key)
../../miniconda3/envs/distil-auto-ml/lib/python3.6/site-packages/pandas/core/indexing.py:1443: in _getitem_tuple
    self._has_valid_tuple(tup)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <pandas.core.indexing._iLocIndexer object at 0x7f94182d7f98>, key = (slice(None, None, None), 0)

    def _has_valid_tuple(self, key: Tuple):
        """
            Check the key for valid keys across my indexer.
            """
        for i, k in enumerate(key):
            if i >= self.ndim:
>               raise IndexingError("Too many indexers")
E               pandas.core.indexing.IndexingError: Too many indexers

../../miniconda3/envs/distil-auto-ml/lib/python3.6/site-packages/pandas/core/indexing.py:700: IndexingError
___________________________________________ TextEncoderPrimitiveTestCase.test_defaults ____________________________________________

self = <test_text_encoder.TextEncoderPrimitiveTestCase testMethod=test_defaults>

    def test_defaults(self) -> None:
        # load test data into a dataframe
        dataset = test_utils.load_dataset(self._dataset_path)
        dataframe = test_utils.get_dataframe(dataset, "learningData")

        # create the encoder
        hyperparams_class = TextEncoderPrimitive.metadata.query()["primitive_code"][
            "class_type_arguments"
        ]["Hyperparams"]
        encoder = TextEncoderPrimitive(hyperparams=hyperparams_class.defaults())
        encoder.set_training_data(
>           inputs=dataframe.iloc[:, [0, 1]], outputs=dataframe.iloc[:, 2]
        )

test/test_text_encoder.py:46:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
distil/primitives/text_encoder.py:124: in set_training_data
    outputs = outputs.iloc[:, 0].values
../../miniconda3/envs/distil-auto-ml/lib/python3.6/site-packages/pandas/core/indexing.py:873: in __getitem__
    return self._getitem_tuple(key)
../../miniconda3/envs/distil-auto-ml/lib/python3.6/site-packages/pandas/core/indexing.py:1443: in _getitem_tuple
    self._has_valid_tuple(tup)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <pandas.core.indexing._iLocIndexer object at 0x7f94182ba778>, key = (slice(None, None, None), 0)

    def _has_valid_tuple(self, key: Tuple):
        """
            Check the key for valid keys across my indexer.
            """
        for i, k in enumerate(key):
            if i >= self.ndim:
>               raise IndexingError("Too many indexers")
E               pandas.core.indexing.IndexingError: Too many indexers

../../miniconda3/envs/distil-auto-ml/lib/python3.6/site-packages/pandas/core/indexing.py:700: IndexingError
======================================================== warnings summary =========================================================
test/test_image_transfer.py::ImageTransferPrimitveTestCase::test_basic
  /home/chris/dev/miniconda3/envs/distil-auto-ml/lib/python3.6/site-packages/torch/nn/functional.py:718: UserWarning: Named tensors and all their associated APIs are an experimental feature and subject to change. Please do not use them for anything important until they are released as stable. (Triggered internally at  /pytorch/c10/core/TensorImpl.h:1156.)
    return torch.max_pool2d(input, kernel_size, stride, padding, dilation, ceil_mode)

-- Docs: http://doc.pytest.org/en/latest/warnings.html
========================================= 6 failed, 32 passed, 1 warnings in 9.36 seconds =========================================
```

--- a/test/audio_dataset_1/datasetDoc.json
+++ b/test/audio_dataset_1/datasetDoc.json
@@ -45,18 +45,12 @@
         },
         {
           "colIndex": 2,
-          "colName": "start",
+          "colName": "start-end-time-slice-of-recording",
           "colType": "real",
           "role": ["boundaryIndicator"]
         },
         {
           "colIndex": 3,
-          "colName": "end",
-          "colType": "real",
-          "role": ["boundaryIndicator"]
-        },
-        {
-          "colIndex": 4,
           "colName": "class",
           "colType": "categorical",
           "role": ["suggestedTarget"]

--- a/test/audio_dataset_1/tables/learningData.csv
+++ b/test/audio_dataset_1/tables/learningData.csv
@@ -1,2 +1,2 @@
-d3mIndex,filename,start,end,class
-0,test_audio.mp3,0.007,0.008,test
+d3mIndex,filename,start-end-time-slice-of-recording,class
+0,test_audio.mp3,"0.007,0.008",test

--- a/test/test_audio_transfer.py
+++ b/test/test_audio_transfer.py
@@ -51,7 +51,7 @@ class AudioTransferPrimitiveTestCase(unittest.TestCase):
         dataframe_primitive = dataset_to_dataframe.DatasetToDataFramePrimitive(
             hyperparams=dataframe_hyperparams_class.defaults()
         )
-        df = dataframe_primitive.produce(inputs=dataset).value
+        audio_df = dataframe_primitive.produce(inputs=dataset).value
 
         audio_df.metadata = audio_df.metadata.add_semantic_type(
             (metadata_base.ALL_ELEMENTS, 0), "http://schema.org/AudioObject"
@@ -73,20 +73,14 @@ class AudioTransferPrimitiveTestCase(unittest.TestCase):
     def test_no_hyperparams_no_semantic_type(self):
         dataset = test_utils.load_dataset(self._dataset_path)
 
-        audio_loader_hyperparams = (
+        auto_loader_hyperparams_class = (
             AudioDatasetLoaderPrimitive.metadata.get_hyperparams()
         )
+        audio_loader_hyperparams = auto_loader_hyperparams_class.defaults()
         audio_loader_primitive = AudioDatasetLoaderPrimitive(
             hyperparams=audio_loader_hyperparams
         )
-        audio_df = audio_loader_primitive.produce(inputs=df).value
-        images.metadata = images.metadata.add_semantic_type(
-            (
-                metadata_base.ALL_ELEMENTS,
-                images.metadata.get_column_index_from_column_name("audio"),
-            ),
-            "http://schema.org/AudioObject",
-        )
+        audio_df = audio_loader_primitive.produce(inputs=dataset).value
 
         audio_transfer_hyperparams = AudioTransferPrimitive.metadata.get_hyperparams()
         primitive_volumes = AudioTransferPrimitive.metadata.get_volumes()

--- a/test/test_text_encoder.py
+++ b/test/test_text_encoder.py
@@ -43,7 +43,7 @@ class TextEncoderPrimitiveTestCase(unittest.TestCase):
         ]["Hyperparams"]
         encoder = TextEncoderPrimitive(hyperparams=hyperparams_class.defaults())
         encoder.set_training_data(
-            inputs=dataframe.iloc[:, [0, 1]], outputs=dataframe.iloc[:, 2]
+            inputs=dataframe.iloc[:, [0, 1]], outputs=dataframe[['bravo']]
         )
         encoder.fit()
         result = encoder.produce(inputs=dataframe).value
@@ -88,7 +88,7 @@ class TextEncoderPrimitiveTestCase(unittest.TestCase):
         ]["Hyperparams"]
         encoder = TextEncoderPrimitive(hyperparams=hyperparams_class.defaults())
         encoder.set_training_data(
-            inputs=dataframe.iloc[:, [0, 1]], outputs=dataframe.iloc[:, 2]
+            inputs=dataframe.iloc[:, [0, 1]], outputs=dataframe[['bravo']]
         )
         encoder.fit()
         result = encoder.produce(inputs=dataframe).value
@@ -118,7 +118,7 @@ class TextEncoderPrimitiveTestCase(unittest.TestCase):
         ]["Hyperparams"]
         encoder = TextEncoderPrimitive(hyperparams=hyperparams_class.defaults())
         encoder.set_training_data(
-            inputs=dataframe.iloc[:, [0, 1]], outputs=dataframe.iloc[:, 2]
+            inputs=dataframe.iloc[:, [0, 1]], outputs=dataframe[['bravo']]
         )
 
         # should fail in this case because we have a label with a cardinality of 1

